sys-kernel/gentoo-kernel: add ~ppc64 keyword to 5.4.61

gyakovlev opened the pull request:
big-endian is not provided by fedora, but we share a keyword.
So I masked it on big-endian profiles, but it's still possible to use savedconfig.

gyakovlev commented:
probably need use.force savedconfig on big-endian.

testing kernel right now.

gyakovlev commented:
ok I think I've caught most breakages and it produced bootable kernel on power9 machine, with external modules.
LMKWDYT

mgorny commented:
I would prefer that you started with 5.8.

gyakovlev commented:
I can only build-test 5.8 but not boot-test on real HW.
But I can give it a spin in a VM.
5.4 is a good target because it's LTS and many prefer it.
testing 5.8 now

mgorny commented:
Covering `src_test` for non-x86 arches would also be nice.

gyakovlev commented:
yeah, I've been experimenting on it a bit and found root-less way to create an raw qemu image from unpacked stage3 directory.
mkfs.ext4 has `-d` flag which takes directory as input and populates fs with it.
so basically
```bash
truncate -s512M image.img
mkfs.ext4 -L gentoo-test -d /path/to/stage3
```
produces a file which can be used as rootfs.
I tried alpine minimal rootfs images (less than 3mb), but they do not contain any kind of initsystem.

our stages are too large...

mgorny commented:
That's nice. I was thinking of writing a minimal pseudo-init that just outputs some string we could catch on console.

gyakovlev commented:
yeah dumping something into it is certanly possible, with one gotcha - it's musl.
maybe we could steal their packages and unpack into fs.
there's also that `sys-process/tini` thing for containers, it's go based but supposed to work inside. but probably not worth it dealing with go.

mgorny commented:
I was saying of making our own minimal image, just `/init` and whatever's needed in `/dev`.

Zlogene commented:
> I would prefer that you started with 5.8.

Rather not, 5.8 is one big glitch.

gyakovlev commented:
ok I found a way, just booted 5.8 in alpine 2.7M image, with no extra PKGs, made using method I described above.
this one
http://dl-cdn.alpinelinux.org/alpine/v3.12/releases/ppc64le/alpine-minirootfs-3.12.0-ppc64le.tar.gz

what we can do is simply add an echo to /etc/inittab inside the image.
there's actually busybox init, which will read it.

```shell-session
qemu-system-ppc64 -m 1G -nographic -nodefaults -serial mon:stdio -no-reboot -kernel /boot/5.8.5/vmlinux -initrd /boot/initramfs-5.8.5.img -hda image.img  -append 'root=LABEL=gentoo-test'            
WARNING: Image format was not specified for 'image.img' and probing guessed raw.                                                                                                                                   
         Automatically detecting the format is dangerous for raw images, write operations on block 0 will be restricted.                                                                                           
         Specify the 'raw' format explicitly to remove the restrictions.                                                                                                                                           
qemu-system-ppc64: warning: TCG doesn't support requested feature, cap-cfpc=workaround                                                                                                                             
qemu-system-ppc64: warning: TCG doesn't support requested feature, cap-sbbc=workaround                                                                                                                             
qemu-system-ppc64: warning: TCG doesn't support requested feature, cap-ibs=workaround                                                                                                                              
qemu-system-ppc64: warning: TCG doesn't support requested feature, cap-ccf-assist=on                                                                                                                               
                                                                                                                                                                                                                   
                                                                                                                                                                                                                   
SLOF **********************************************************************                                                                                                                                        
QEMU Starting                                                                                                                                                                                                      
 Build Date = Jul 17 2020 11:15:24                                                                                                                                                                                 
 FW Version = git-e18ddad8516ff2cf                                                                                                                                                                                 
 Press "s" to enter Open Firmware.                                                                                                                                                                                 
                                                                                                                                                                                                                   
Populating /vdevice methods                                                                                                                                                                                        
Populating /vdevice/vty@71000000                                                                                                                                                                                   
Populating /vdevice/nvram@71000001                                                                                                                                                                                 
Populating /vdevice/v-scsi@71000002                                                                                                                                                                                
       SCSI: Looking for devices                                                                                                                                                                                   
          8000000000000000 DISK     : "QEMU     QEMU HARDDISK    2.5+"                                                                                                                                             
Populating /pci@800000020000000                                                                                                                                                                                    
No NVRAM common partition, re-initializing...                                                                                                                                                                      
Scanning USB                                                                                                                                                                                                       
Using default console: /vdevice/vty@71000000                                                                                                                                                                       
Detected RAM kernel at 400000 (1e5ae24 bytes)                                                                                                                                                                      
                                                                                                                                                                                                                   
  Welcome to Open Firmware                                                                                                                                                                                         
                                                                                                                                                                                                                   
  Copyright (c) 2004, 2017 IBM Corporation All rights reserved.                                                                                                                                                    
  This program and the accompanying materials are made available                                                                                                                                                   
  under the terms of the BSD License available at                                                                                                                                                                  
  http://www.opensource.org/licenses/bsd-license.php                                                                                                                                                               
                                                                                                                                                                                                                   
Booting from memory...                                                                                                                                                                                             
OF stdout device is: /vdevice/vty@71000000                                                                                                                                                                         
Preparing to boot Linux version 5.8.5 (portage@localhost) (powerpc64le-unknown-linux-gnu-gcc (Gentoo 9.3.0-r1 p3) 9.3.0, GNU ld (Gentoo 2.34 p6) 2.34.0) #1 SMP Fri Aug 28 00:02:22 PDT 2020                       
Detected machine type: 0000000000000101                                                                                                                                                                            
command line: root=LABEL=gentoo-test                                                                                                                                                                               
Max number of cores passed to firmware: 2048 (NR_CPUS = 2048)                                                                                                                                                      
Calling ibm,client-architecture-support... done                                                                                                                                                                    
memory layout at init:
  memory_limit : 0000000000000000 (16 MB aligned)                                                                                                                                                                  
  alloc_bottom : 0000000002d80000                                                                                                                                                                                  
  alloc_top    : 0000000030000000                                                                                                                                                                                  
  alloc_top_hi : 0000000040000000                                                                                                                                                                                  
  rmo_top      : 0000000030000000                                                                                                                                                                                  
  ram_top      : 0000000040000000                                                                                                                                                                                  
instantiating rtas at 0x000000002fff0000... done                                                                                                                                                                   
prom_hold_cpus: skipped                                                                                                                                                                                            
copying OF device tree...                                                                                                                                                                                          
Building dt strings...                                                                                                                                                                                             
Building dt structure...                                                                                                                                                                                           
Device tree strings 0x0000000002d90000 -> 0x0000000002d90aa6                                                                                                                                                       
Device tree struct  0x0000000002da0000 -> 0x0000000002db0000                                                                                                                                                       
Quiescing Open Firmware ...                                                                                                                                                                                        
Booting Linux via __start() @ 0x0000000000400000 ...                                                                                                                                                               
[    0.000000] radix-mmu: Page sizes from device-tree:                                                                                                                                                             
[    0.000000] radix-mmu: Page size shift = 12 AP=0x0                                                                                                                                                              
[    0.000000] radix-mmu: Page size shift = 16 AP=0x5                                                                                                                                                              
[    0.000000] radix-mmu: Page size shift = 21 AP=0x1                                                                                                                                                              
[    0.000000] radix-mmu: Page size shift = 30 AP=0x2                                                                                                                                                              
[    0.000000] radix-mmu: Activating Kernel Userspace Execution Prevention                                                                                                                                         
[    0.000000] radix-mmu: Activating Kernel Userspace Access Prevention                                                                                                                                            
[    0.000000] radix-mmu: Mapped 0x0000000000000000-0x0000000040000000 with 1.00 GiB pages (exec)                                                                                                                  
[    0.000000] lpar: Using radix MMU under hypervisor                                                                                                                                                              
[    0.000000] Linux version 5.8.5 (portage@localhost) (powerpc64le-unknown-linux-gnu-gcc (Gentoo 9.3.0-r1 p3) 9.3.0, GNU ld (Gentoo 2.34 p6) 2.34.0) #1 SMP Fri Aug 28 00:02:22 PDT 2020                          
[    0.000000] Found initrd at 0xc000000002270000:0xc000000002d734ae                                                                                                                                               
[    0.000000] Using pSeries machine description                                                                                                                                                                   
[    0.000000] printk: bootconsole [udbg0] enabled                                                                                                                                                                 
[    0.000000] Partition configured for 1 cpus.                                                                                                                                                                    
[    0.000000] CPU maps initialized for 1 thread per core
[    0.000000] phys_mem_size     = 0x40000000                                                                                                                                                                      
[    0.000000] dcache_bsize      = 0x80                                                                                                                                                                            
[    0.000000] icache_bsize      = 0x80                                                                                                                                                                            
[    0.000000] cpu_features      = 0x0001c06f8f4f91a7                                                                                                                                                              
[    0.000000]   possible        = 0x0007fbffcf5fb1a7                                                                                                                                                              
[    0.000000]   always          = 0x00000003800081a1                                                                                                                                                              
[    0.000000] cpu_user_features = 0xdc0065c2 0xaee00000                                                                                                                                                           
[    0.000000] mmu_features      = 0xbc006041                                                                                                                                                                      
[    0.000000] firmware_features = 0x00000085455a445f                                                                                                                                                              
[    0.000000] vmalloc start     = 0xc008000000000000                                                                                                                                                              
[    0.000000] IO start          = 0xc00a000000000000                                                                                                                                                              
[    0.000000] vmemmap start     = 0xc00c000000000000                                                                                                                                                              
[    0.000000] -----------------------------------------------------
[    0.000000] numa:   NODE_DATA [mem 0x3ffc4900-0x3ffcbfff]
[    0.000000] rfi-flush: fallback displacement flush available
[    0.000000] rfi-flush: ori type flush available
[    0.000000] rfi-flush: mttrig type flush available
[    0.000000] link-stack-flush: software flush enabled.
[    0.000000] count-cache-flush: hardware assisted flush sequence enabled
[    0.000000] stf-barrier: eieio barrier available
[    0.000000] PCI host bridge /pci@800000020000000  ranges:
[    0.000000]   IO 0x0000200000000000..0x000020000000ffff -> 0x0000000000000000
[    0.000000]  MEM 0x0000200080000000..0x00002000ffffffff -> 0x0000000080000000 
[    0.000000]  MEM 0x0000210000000000..0x000021ffffffffff -> 0x0000210000000000 
[    0.000000] PCI: OF: PROBE_ONLY disabled
[    0.000000] PPC64 nvram contains 65536 bytes
[    0.000000] barrier-nospec: using ORI speculation barrier
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x000000003fffffff]
[    0.000000]   Device   empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x000000003fffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x000000003fffffff]
[    0.000000] percpu: Embedded 11 pages/cpu s624920 r0 d95976 u720896
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 16368
[    0.000000] Policy zone: Normal
[    0.000000] Kernel command line: root=LABEL=gentoo-test
[    0.000000] Dentry cache hash table entries: 131072 (order: 4, 1048576 bytes, linear)
[    0.000000] Inode-cache hash table entries: 65536 (order: 3, 524288 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 918528K/1048576K available (16064K kernel code, 2176K rwdata, 4736K rodata, 6208K init, 1835K bss, 130048K reserved, 0K cma-reserved)
[    0.000000] random: get_random_u64 called from __kmem_cache_create+0x58/0x760 with crng_init=0
[    0.000000] SLUB: HWalign=128, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] ftrace: allocating 38651 entries in 15 pages
[    0.000000] ftrace: allocated 15 pages with 4 groups
[    0.000000] rcu: Hierarchical RCU implementation. 
[    0.000000] rcu:     RCU restricting CPUs from NR_CPUS=2048 to nr_cpu_ids=1.
[    0.000000]  Trampoline variant of Tasks RCU enabled.
[    0.000000]  Rude variant of Tasks RCU enabled.
[    0.000000]  Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.                                                                                                                              
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=1                                                                                                                                        
[    0.000000] NR_IRQS: 512, nr_irqs: 512, preallocated irqs: 16                                                                                                                                                   
[    0.000000] xive: Using IRQ range [0-0]                                                                                                                                                                         
[    0.000000] xive: Interrupt handling initialized with spapr backend                                                                                                                                             
[    0.000000] xive: Using priority 6 for all interrupts                                                                                                                                                           
[    0.000000] xive: Using 64kB queues                                                                                                                                                                             
[    0.000000] rcu:     Offload RCU callbacks from CPUs: (none).                                                                                                                                                   
[    0.000156] time_init: 56 bit decrementer (max: 7fffffffffffff)                                                                                                                                                 
[    0.000820] clocksource: timebase: mask: 0xffffffffffffffff max_cycles: 0x761537d007, max_idle_ns: 440795202126 ns                                                                                              
[    0.001572] clocksource: timebase mult[1f40000] shift[24] registered                                                                                                                                            
[    0.009200] Console: colour dummy device 80x25                                                                                                                                                                  
[    0.010300] printk: console [hvc0] enabled                                                                                                                                                                      
[    0.010300] printk: console [hvc0] enabled                                                                                                                                                                      
[    0.010846] printk: bootconsole [udbg0] disabled                                                                                                                                                                
[    0.010846] printk: bootconsole [udbg0] disabled                                                                                                                                                                
[    0.018280] pid_max: default: 32768 minimum: 301                                                                                                                                                                
[    0.019904] LSM: Security Framework initializing                                                                                                                                                                
[    0.021650] Yama: becoming mindful.                                                                                                                                                                             
[    0.023123] SELinux:  Initializing.                                                                                                                                                                             
[    0.025292] Mount-cache hash table entries: 8192 (order: 0, 65536 bytes, linear)                                                                                                                                
[    0.025390] Mountpoint-cache hash table entries: 8192 (order: 0, 65536 bytes, linear)
[    0.085488] EEH: pSeries platform initialized
[    0.086275] POWER9 performance monitor hardware support registered
[    0.087699] rcu: Hierarchical SRCU implementation.
[    0.103353] smp: Bringing up secondary CPUs ...
[    0.103529] smp: Brought up 1 node, 1 CPU
[    0.103927] numa: Node 0 CPUs: 0
[    0.104145] Using standard scheduler topology
[    0.120789] devtmpfs: initialized
[    0.151334] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.151584] futex hash table entries: 256 (order: -1, 32768 bytes, linear)
[    0.156398] xor: measuring software checksum speed
[    0.254444]    8regs     :  2252.800 MB/sec
[    0.387193]    8regs_prefetch:  1830.400 MB/sec
[    0.488071]    32regs    :  2208.000 MB/sec
[    0.588984]    32regs_prefetch:  1817.600 MB/sec
[    0.689804]    altivec   :  1632.000 MB/sec
[    0.689924] xor: using function: 8regs (2252.800 MB/sec)
[    0.692835] thermal_sys: Registered thermal governor 'fair_share'
[    0.693011] thermal_sys: Registered thermal governor 'step_wise'
[    0.704117] NET: Registered protocol family 16
[    0.709823] audit: initializing netlink subsys (disabled)
[    0.717219] cpuidle: using governor menu
[    0.720777] audit: type=2000 audit(1598600916.660:1): state=initialized audit_enabled=0 res=1
[    0.723326] pstore: Registered nvram as persistent store backend
Linux ppc64le
#1 SMP Fri Aug 2[    0.745072] PCI: Probing PCI hardware
[    0.747538] PCI host bridge to bus 0000:00
[    0.748280] pci_bus 0000:00: root bus resource [io  0x10000-0x1ffff] (bus address [0x0000-0xffff])
[    0.748460] pci_bus 0000:00: root bus resource [mem 0x200080000000-0x2000ffffffff] (bus address [0x80000000-0xffffffff])
[    0.748539] pci_bus 0000:00: root bus resource [mem 0x210000000000-0x21ffffffffff]
[    0.748875] pci_bus 0000:00: root bus resource [bus 00-ff]
[    0.753751] IOMMU table initialized, virtual merging enabled
[    0.755789] pci_bus 0000:00: resource 4 [io  0x10000-0x1ffff]
[    0.755849] pci_bus 0000:00: resource 5 [mem 0x200080000000-0x2000ffffffff]
[    0.755895] pci_bus 0000:00: resource 6 [mem 0x210000000000-0x21ffffffffff]
[    0.756173] EEH: No capable adapters found: recovery disabled.
[    0.801575] HugeTLB registered 2.00 MiB page size, pre-allocated 0 pages
[    0.801657] HugeTLB registered 1.00 GiB page size, pre-allocated 0 pages
[    2.422902] alg: No test for 842 (842-generic)
[    2.423420] alg: No test for 842 (842-scomp)
[    2.476025] raid6: skip pq benchmark and using algorithm vpermxor8
[    2.476186] raid6: using intx1 recovery algorithm 
[    2.478627] iommu: Default domain type: Translated 
[    2.483919] vgaarb: loaded
[    2.486870] SCSI subsystem initialized
[    2.490402] usbcore: registered new interface driver usbfs
[    2.490886] usbcore: registered new interface driver hub
[    2.491306] usbcore: registered new device driver usb
[    2.492231] pps_core: LinuxPPS API ver. 1 registered
[    2.492280] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    2.492443] PTP clock support registered
[    2.493764] EDAC MC: Ver: 3.0.0
[    2.509658] NetLabel: Initializing
[    2.509720] NetLabel:  domain hash size = 128
[    2.509758] NetLabel:  protocols = UNLABELED CIPSOv4 CALIPSO
[    2.511015] NetLabel:  unlabeled traffic allowed by default
[    2.516823] clocksource: Switched to clocksource timebase
[    2.803750] VFS: Disk quotas dquot_6.6.0
[    2.804154] VFS: Dquot-cache hash table entries: 8192 (order 0, 65536 bytes)
[    2.839163] NET: Registered protocol family 2
[    2.848172] tcp_listen_portaddr_hash hash table entries: 4096 (order: 0, 65536 bytes, linear)
[    2.848454] TCP established hash table entries: 8192 (order: 0, 65536 bytes, linear)
[    2.848751] TCP bind hash table entries: 8192 (order: 1, 131072 bytes, linear)
[    2.849098] TCP: Hash tables configured (established 8192 bind 8192)
[    2.851784] UDP hash table entries: 2048 (order: 0, 65536 bytes, linear)
[    2.852159] UDP-Lite hash table entries: 2048 (order: 0, 65536 bytes, linear)
[    2.854924] NET: Registered protocol family 1
[    2.855412] NET: Registered protocol family 44
[    2.855688] PCI: CLS 0 bytes, default 128
[    2.860168] Trying to unpack rootfs image as initramfs...
[    3.802677] Freeing initrd memory: 11264K
[    3.806858] rtas_flash: no firmware flash support 
[    3.820175] Initialise system trusted keyrings
[    3.822932] Key type blacklist registered
[    3.824049] workingset: timestamp_bits=38 max_order=14 bucket_order=0
[    3.848821] zbud: loaded
[    3.893659] NET: Registered protocol family 38
[    3.894063] Key type asymmetric registered
[    3.894194] Asymmetric key parser 'x509' registered
[    3.894522] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 245)
[    3.895810] io scheduler mq-deadline registered
[    3.896060] io scheduler kyber registered
[    3.897010] io scheduler bfq registered
[    3.905689] atomic64_test: passed
[    3.906870] shpchp: Standard Hot Plug PCI Controller Driver version: 0.4
[    3.915505] Serial: 8250/16550 driver, 32 ports, IRQ sharing enabled
[    3.951350] Non-volatile memory driver v1.3
[    3.960563] libphy: Fixed MDIO Bus: probed
[    3.962770] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    3.963128] ehci-pci: EHCI PCI platform driver
[    3.963498] ohci_hcd: USB 1.1 'Open' Host Controller (OHCI) Driver
[    3.963799] ohci-pci: OHCI PCI platform driver
[    3.964145] uhci_hcd: USB Universal Host Controller Interface driver
[    3.965235] usbcore: registered new interface driver usbserial_generic
[    3.965669] usbserial: USB Serial support registered for generic
[    3.967239] mousedev: PS/2 mouse device common for all mice
[    3.971018] rtc-generic rtc-generic: registered as rtc0
[    3.972095] rtc-generic rtc-generic: setting system clock to 2020-08-28T07:48:40 UTC (1598600920)
[    3.973255] device-mapper: uevent: version 1.0.3
[    3.974754] device-mapper: ioctl: 4.42.0-ioctl (2020-02-27) initialised: dm-devel@redhat.com
[    3.978873] hid: raw HID events driver (C) Jiri Kosina
[    3.979974] usbcore: registered new interface driver usbhid
[    3.980029] usbhid: USB HID core driver
[    3.980655] drop_monitor: Initializing network drop monitor service
[    3.982233] Initializing XFRM netlink socket
[    3.984508] NET: Registered protocol family 10
[    4.315071] Segment Routing with IPv6
[    4.317037] RPL Segment Routing with IPv6
[    4.320161] mip6: Mobile IPv6
[    4.320599] NET: Registered protocol family 17
[    4.325603] drmem: No dynamic reconfiguration memory found
[    4.358411] registered taskstats version 1
[    4.359865] Loading compiled-in X.509 certificates
[    4.363685] zswap: loaded using pool lzo/zbud
[    4.366839] Key type ._fscrypt registered
[    4.366919] Key type .fscrypt registered
[    4.366964] Key type fscrypt-provisioning registered
[    4.375600] Btrfs loaded, crc32c=crc32c-generic
[    4.377788] pstore: Using crash dump compression: deflate
[    4.515961] Key type encrypted registered
[    4.517137] ima: No TPM chip found, activating TPM-bypass!
[    4.517433] ima: Allocated hash algorithm: sha256
[    4.521029] ima: No architecture policies found
[    4.686985] Freeing unused kernel memory: 6208K
[    4.687095] This architecture does not have kernel memory protection.
[    4.687647] Run /init as init process
[    7.804810] dracut: ZFS: Using hostid from /etc/hostid: d3ccc164
[    8.043919] systemd-udevd[242]: starting version 3.2.9
[    8.057942] random: systemd-udevd: uninitialized urandom read (16 bytes read)
[    8.061892] random: systemd-udevd: uninitialized urandom read (16 bytes read)
[    8.063287] random: systemd-udevd: uninitialized urandom read (16 bytes read)
[    8.102552] udevd[243]: starting eudev-3.2.9
[    9.564680] urandom_read: 1 callbacks suppressed
[    9.564938] random: systemd-udevd: uninitialized urandom read (16 bytes read)
[    9.565406] random: systemd-udevd: uninitialized urandom read (16 bytes read)
[    9.884509] synth uevent: /devices/vio: failed to send uevent
[    9.885004] vio vio: uevent: failed to send synthetic uevent
[    9.932416] random: systemd-udevd: uninitialized urandom read (16 bytes read)
```

mgorny commented:
Could you make a patch, ideally for all arches?

gyakovlev commented:
> Could you make a patch, ideally for all arches?

that's the plan!
will take some time poking though.

gyakovlev commented:
ok `::sysinit:/bin/echo "KERNEL BOOTED"` in inittab worked =)

```
Booting from memory...
OF stdout device is: /vdevice/vty@71000000
Preparing to boot Linux version 5.8.5 (portage@localhost) (powerpc64le-unknown-linux-gnu-gcc (Gentoo 9.3.0-r1 p3) 9.3.0, GNU ld (Gentoo 2.34 p6) 2.34.0) #1 SMP Fri Aug 28 00:02:22 PDT 2020
Detected machine type: 0000000000000101
command line: root=/dev/sda rootfstype=ext4 rootdelay=5 rd.hostonly=0 quiet rd.driver.pre=ahci_platform
Max number of cores passed to firmware: 2048 (NR_CPUS = 2048)
Calling ibm,client-architecture-support... done
memory layout at init:
  memory_limit : 0000000000000000 (16 MB aligned)
  alloc_bottom : 0000000004520000
  alloc_top    : 0000000030000000
  alloc_top_hi : 0000000040000000
  rmo_top      : 0000000030000000
  ram_top      : 0000000040000000
instantiating rtas at 0x000000002fff0000... done
prom_hold_cpus: skipped
copying OF device tree...
Building dt strings...
Building dt structure...
Device tree strings 0x0000000004530000 -> 0x0000000004530aa6
Device tree struct  0x0000000004540000 -> 0x0000000004550000
Quiescing Open Firmware ...
Booting Linux via __start() @ 0x0000000000400000 ...
Linux ppc64le
KERNEL BOOTED
```

we can use this.

mgorny commented on sys-kernel/gentoo-kernel/gentoo-kernel-5.4.61.ebuild:
```diff
@@ -24,14 +24,18 @@ SRC_URI+=" https://cdn.kernel.org/pub/linux/kernel/v$(ver_cut 1).x/${MY_P}.tar.x
 	)
+	ppc64? (
+		https://src.fedoraproject.org/rpms/kernel/raw/${CONFIG_HASH}/f/kernel-ppc64le.config
+			-> kernel-ppc64.config.${CONFIG_VER}
```
Please keep the `le` for clarity.

gyakovlev replied:
done, fixed filename and pushed.

mgorny reviewed (commented):
Cool, thanks! Feel free to work separately on that, and commits the ppc64 support, modulo this one comment.
